components/Layout.jsx

```diff
@@ -1,9 +1,15 @@
-import Header from "./Header"
 import Footer from "./Footer"
+import Header from "./Header"
+
+import { useRouter } from "next/router"
+import { locales } from "../data/locales"
 
 export const Layout = ({ children }) => {
+  const { locale } = useRouter()
+  const dir = locales.find((l) => l.code === locale)?.dir || "ltr"
+
   return (
-    <div className="mx-auto max-w-site px-6 lg:px-16">
+    <div dir={dir}>
```

Comment:
ideally this `dir` attribute would go on the html, but c'est la vie?

Reply:
maybe we could grab the locale from `ctx.locale` like this https://github.com/vercel/next.js/discussions/19049#discussioncomment-1322838?

Reply:
ohhhhh i see, i forgot this needs to handle the front end routing too :0 do u think it makes more sense to have the `useeffect` or just keep the `dir` in the Layout as it was before? I'm just unsure of what the best practice is here

Reply:
no styling issues unless we needed to style the html/body conditionally, and AFAICT no impact on SEO, accessibility, etc, but semantically it seems odd to separate it from the `lang` attribute that gets put on the html element automatically.

[w3c does seem prescriptive about it being on the html element](https://www.w3.org/International/questions/qa-html-dir#quickanswer), so maybe we do that

next has always been a pain about managing the html element, tho. wish it had the same toolset as react-head for that

Reply:
ah oki, let's just keep it on the html element for now then